### PR TITLE
[WIP] Update various errors/typos in rst documentation and docstrings

### PR DESCRIPTION
**Pull request by lewoudar**
Hi folks,
This PR fixes some typos in documentation. I marked it WIP because I want another change but I don't know how to do it. It concerns this section in reference-io
subprocess

I don't know if is possible to create an alias for a `:ref:` :(

### CODE REVIEW

File: trio/_core/_generated_run.py [hunk 1]
```diff
@@ -193,21 +193,21 @@ def current_trio_token():
...
                      assert lock.locked()
-                     assert lock._owner is trio.current_task()
+                     assert lock._owner is trio.hazmat.current_task()
                      lock.release()
                      try:
                          # The child has a prior claim, so we can't have it
                          lock.acquire_nowait()
                      except trio.WouldBlock:
-                         assert lock._owner is child
+                         assert lock._owner is not trio.hazmat.current_task()
```
Comment (njsmith) [inline trio/_core/_generated_run.py]: Unfortunately, this file is automatically generated code... the real docstring is in `trio/_core/_run.py`. So the process to fix it is:

1. make the edits to `trio/_core/_run.py`
2. run `python trio/_tools/gen_exports.py` to regenerate this file
3. commit the changes to both `trio/_core/_run.py` and `trio/_core/_generated_run.py`

Comment (lewoudar) [reply trio/_core/_generated_run.py]: Oops, I missed that information 😆 OK I changed it

**Comment by pquentin** [comment]
Cycling to fix the Travis check.

File: trio/_highlevel_ssl_helpers.py [hunk 2]
```diff
@@ -84,7 +84,7 @@
       host (str, bytes, or None): The address to bind to; use ``None`` to bind
           to the wildcard address. See :func:`open_tcp_listeners`.
       https_compatible (bool): See :class:`~trio.SSLStream` for details.
-      backlog (int or None): See :class:`~trio.SSLStream` for details.
+      backlog (int or None): See :func:`open_tcp_listeners` for details.
```
Comment (pquentin) [inline trio/_highlevel_ssl_helpers.py]: nitpick: I think there's another occurrence of this in `trio.serve_ssl_over_tcp`

Comment (lewoudar) [reply trio/_highlevel_ssl_helpers.py]: Ooh I didn't see that, do you want that I make another PR?

Comment (pquentin) [reply trio/_highlevel_ssl_helpers.py]: @lewoudar Yes, that would be great. 👍

**Pull request merged**